pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -749,31 +743,49 @@ func (ku *Upgrader) copyCustomPropertiesToNewNode(client armhelpers.KubernetesCl
 				continue
 			}
 
-			err = ku.copyCustomNodeProperties(client, oldNodeName, oldNode, newNodeName, newNode)
+			copiedProps, err := ku.copyCustomNodeProperties(client, oldNodeName, oldNode, newNodeName, newNode)
 			if err != nil {
 				ku.logger.Debugf("Failed to copy custom annotations, labels, taints from old node %s to new node %s: %v", oldNodeName, newNodeName, err)
 				time.Sleep(time.Second * 5)
 			} else {
-				ch <- struct{}{}
+				ch <- copiedProps
 			}
 		}
 	}()
 
-	for {
-		select {
-		case <-ch:
-			ku.logger.Infof("Successfully copied custom annotations, labels, taints from old node %s to new node %s.", oldNodeName, newNodeName)
-			return nil
-		case <-time.After(nodePropertiesCopyTimeout):
-			err := fmt.Errorf("Copying custom annotations, labels, taints from old node %s to new node %s can't complete within %v", oldNodeName, newNodeName, nodePropertiesCopyTimeout)
-			ku.logger.Errorf(err.Error())
-			return err
-		}
+	copiedProps := false
+	select {
+	case copiedProps = <-ch:
+	case <-time.After(nodePropertiesCopyTimeout):
+		err := fmt.Errorf("Copying custom annotations, labels, taints from old node %s to new node %s can't complete within %v", oldNodeName, newNodeName, nodePropertiesCopyTimeout)
+		ku.logger.Errorf(err.Error())
+		return err
 	}
+
+	if !copiedProps {
+		ku.logger.Infof("Successfully copied custom annotations, labels, taints from old node %s to new node %s.", oldNodeName, newNodeName)
+		return nil
+	}
+
+	ku.logger.Infof("Successfully copied custom annotations, labels, taints from old node %s to new node %s.", oldNodeName, newNodeName)
+
+	var cordonDrainTimeout time.Duration
+	if ku.cordonDrainTimeout == nil {
+		cordonDrainTimeout = defaultCordonDrainTimeout
+	} else {
+		cordonDrainTimeout = *ku.cordonDrainTimeout
+	}
+	err := operations.JustDrainNodeWithClient(client, ku.logger, newNodeName, cordonDrainTimeout)
+	if err != nil {
+		ku.logger.Warningf("Error draining agent VM %s. Proceeding with copying node properties. Error: %v", newNodeName, err)
+		//would like to return err but draining is even more likely to fail
```

Comment:
anyone want to vote for erroring out here?

Reply:
In aks-engine I have a work item to add an agentpool setting on when to fail based on drain.